src/snowflake/connector/pandas_tools.py

```diff
@@ -178,6 +183,54 @@ def write_pandas(
     else:
         columns = ",".join(list(df.columns))
 
+    if auto_create_table:
+        file_format_name = None
+        while True:
+            try:
+                file_format_name = (
+                    '"'
+                    + "".join(random.choice(string.ascii_lowercase) for _ in range(5))
+                    + '"'
+                )
+                file_format_sql = (
+                    f"CREATE FILE FORMAT {file_format_name} "
+                    f"/* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+                    f"TYPE=PARQUET COMPRESSION={compression_map[compression]}"
+                )
+                logger.debug("creating file format with '{}'".format(file_format_sql))
+                cursor.execute(file_format_sql, _is_internal=True).fetchall()
+                break
+            except ProgrammingError as pe:
+                if pe.msg.endswith("already exists."):
+                    continue
+                raise
+        infer_schema_sql = f"SELECT COLUMN_NAME, TYPE FROM table(infer_schema(location=>'@\"{stage_name}\"', file_format=>'{file_format_name}'))"
+        logger.debug(f"inferring schema with {infer_schema_sql}")
+        column_type_mapping = dict(
+            cursor.execute(infer_schema_sql, _is_internal=True).fetchall()
+        )
+        # Infer schema can return the columns out of order depending on the chunking we do when uploading
+        # so we have to iterate through the dataframe columns to make sure we create the table with its
+        # columns in order
+        if quote_identifiers:
+            create_table_columns = ", ".join(
+                [f'"{c}" {column_type_mapping[c]}' for c in df.columns]
+            )
+        else:
+            create_table_columns = ", ".join(
+                [f"{c} {column_type_mapping[c]}" for c in df.columns]
+            )
+        create_table_sql = (
+            f"CREATE {'TEMP ' if temp_table else ''}TABLE IF NOT EXISTS {location} "
+            f"({create_table_columns})"
+            f"/* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+        )
+        logger.debug(f"auto creating table with {create_table_sql}")
```

Comment:
Keeping consistent with already existing logging messages
```suggestion
        logger.debug(f"auto creating table with '{create_table_sql}'")
```